core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -234,6 +234,11 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
 
         @Override
+        public String describeTasks(List<ShardRoutingEntry> tasks) {
+            return tasks.stream().map(entry -> entry.getShardRouting().toString()).reduce((s1, s2) -> s1 + ", " + s2).orElse("");
```

Comment:
Maybe, but chasing things down in logs is no fun (e.g., if this task summary shows up in the slow task log).